action.yml

```
@@ -0,0 +1,232 @@
+name: Run tests
+description: Run tests for a single matrix entry
+
+inputs:
+  endtoend:
+    type: boolean
+    default: false
+  endtoend_suite:
+    type: string
+    required: false
+    default: ''
+  endtoend_config:
+    type: string
+    required: false
+    default: ''
+  phpcoverage:
+    type: boolean
+    default: false
+  phplinting:
+    type: boolean
+    default: false
+  phpunit:
+    type: boolean
+    default: false
+  phpunit_suite:
+    type: string
+    required: false
+    default: ''
+  phpunit_fail_on_warning:
+    type: boolean
+    default: false
+  js:
+    type: boolean
+    default: false
+
+runs:
+  using: composite
+  steps:
+
+    - name: Validate inputs
+      shell: bash
+      env:
+        PHPUNIT_SUITE: ${{ inputs.phpunit_suite }}
+        ENDTOEND_SUITE: ${{ inputs.endtoend_suite }}
+        ENDTOEND_CONFIG: ${{ inputs.endtoend_config }}
+      run: |
+        if ! [[ "$PHPUNIT_SUITE" =~ ^[a-zA-Z0-9_\-]*$ ]]; then
+          echo "Invalid input for phpunit_suite"
+          exit 1
+        fi
+        if ! [[ "$ENDTOEND_SUITE" =~ ^[a-zA-Z0-9_\-]*$ ]]; then
+          echo "Invalid input for endtoend_suite"
+          exit 1
+        fi
+        if ! [[ "$ENDTOEND_CONFIG" =~ ^[a-zA-Z0-9_\./\-]*$ ]]; then
+          echo "Invalid input for endtoend_config"
+          exit 1
+        fi
+
+    - name: Run PHPUnit
+      # input booleans are converted to strings
+      # https://docs.github.com/en/actions/learn-github-actions/expressions#functions
```

We're actually explicitly converting it to a string in the matrix anyway, so even if this changed in github actions, the string comparison would be correct.
```suggestion
      # input booleans are converted to strings
```

---

Updated